src/server/pfs/server/driver.go

```diff
@@ -776,32 +794,14 @@ func (d *driver) propagateBranches(txnCtx *txncontext.TransactionContext, branch
 		return len(subvBIs[i].Provenance) < len(subvBIs[j].Provenance)
 	})
 
-	// Iterate through downstream branches and determine which need a new commit.
+	// Iterate through downstream branches and add new commits.
 	hasNewCommits := false
 	for _, subvBI := range subvBIs {
 		// Do not propagate an open commit onto spout output branches (which should
 		// only have a single provenance on a spec commit)
 		if len(subvBI.Provenance) == 1 && subvBI.Provenance[0].Repo.Type == pfs.SpecRepoType {
 			continue
 		}
-
-		// We need to create new commits or aliases if any of this branch and its
-		// provenances disagree on their commitset.
-		ids := []string{subvBI.Head.ID}
-		for _, provOfSubvB := range subvBI.Provenance {
-			provOfSubvBI, err := getBranchInfo(provOfSubvB)
-			if err != nil {
-				return err
-			}
-			ids = append(ids, provOfSubvBI.Head.ID)
-		}
-
-		if allSameString(ids) {
-			if ids[0] == txnCtx.CommitSetID {
-				hasNewCommits = true
-			}
-			continue
```

Comment:
I may be mis-reading something, but in the case that we only update a branch by starting/stopping a pipeline, wouldn't  this `continue` stop `propagateBranches` from creating a new commit/job?

Reply:
in the old code, if we stop a pipeline, or start one while everything in provenance is still up to date, yes, this skips the commit/job. However, any downstream branches would still hit this check

Reply:
thank you for explaining to me over voice :)